sycl/source/detail/scheduler/commands.cpp

```diff
@@ -1724,12 +1724,18 @@ pi_result ExecCGCommand::SetKernelParamsAndLaunch(
 void DispatchNativeKernel(void *Blob) {
   // First value is a pointer to Corresponding CGExecKernel object.
   CGExecKernel *HostTask = *(CGExecKernel **)Blob;
+  bool MayDeleteCG = static_cast<bool>(*((void **)Blob + 1));
 
   // Other value are pointer to the buffers.
-  void **NextArg = (void **)Blob + 1;
+  void **NextArg = (void **)Blob + 2;
```

Comment:
While you are here, DRY and do not use blurry C casts:
```suggestion
  auto NextArg = static_cast<void **>(Blob) + 2;
```